sources/packages/api/src/services/designation-agreement/designation-agreement.service.ts

```diff
@@ -147,4 +163,114 @@ export class DesignationAgreementService extends RecordDataModelService<Designat
+   * @param designationId
+   * @returns flag that indicates if the designationId is valid data.
+   */
+  async designationForUpdateExist(designationId: number): Promise<boolean> {
+    const found = await this.repo
+      .createQueryBuilder("designation")
+      .select("1")
+      .where("designation.id = :designationId", { designationId })
+      .andWhere("designation.designationStatus != :designationStatus", {
+        designationStatus: DesignationAgreementStatus.Declined,
+      })
+      .getRawOne();
+    return !!found;
+  }
+
+  /**
+   * Update designation for Approval/Denial or re-approve.
+   * @param designationId Designation which is going to be updated.
+   * @param userId User who updates the designation.
+   * @param designationPayload update payload.
+   */
+  async updateDesignation(
+    designationId: number,
+    userId: number,
+    designationPayload: UpdateDesignationDto,
+  ): Promise<void> {
+    const designation = new DesignationAgreement();
+    designation.id = designationId;
+    designation.designationStatus = designationPayload.designationStatus;
+    designation.startDate = designationPayload.startDate;
+    designation.endDate = designationPayload.endDate;
+    designation.assessedBy = { id: userId } as User;
+    designation.assessedDate = new Date();
+    designation.designationAgreementLocations =
+      designationPayload.locationsDesignations?.map(
+        (locationPayload: UpdateDesignationLocationDto) => {
+          const location = new DesignationAgreementLocation();
+          location.id = locationPayload.designationLocationId;
+          location.requested = locationPayload.requested;
+          location.approved = locationPayload.approved;
+          location.institutionLocation = {
+            id: locationPayload.locationId,
+          } as InstitutionLocation;
+          location.creator = !locationPayload.designationLocationId
+            ? ({ id: userId } as User)
+            : undefined;
+          location.modifier = { id: userId } as User;
+          return location;
+        },
+      );
+    return this.connection.transaction(async (transactionalEntityManager) => {
+      await transactionalEntityManager
+        .getRepository(DesignationAgreement)
+        .save(designation);
+
+      const note = new Note();
+      note.noteType = NoteType.Designation;
+      note.description = designationPayload.note;
+      note.creator = { id: userId } as User;
+
+      const updateNote = await transactionalEntityManager
+        .getRepository(Note)
+        .save(note);
+
+      await transactionalEntityManager
+        .getRepository(Institution)
+        .createQueryBuilder()
+        .relation(Institution, "notes")
+        .of({ id: designationPayload.institutionId } as Institution)
+        .add(updateNote);
+    });
+  }
+  /**
+   * Private service method that retrieves designation summary
+   * based on the given filter.
+   * @param filterColumn
```

Comment:
what is the difference between filter and search here?

Reply:
Added comments to explain as discussed.